### PR TITLE
Clean Up Webpack

### PR DESCRIPTION
**Pull request by swashcap**
Asana task: https://app.asana.com/0/45796090191289/166361544315537
- **Problem:**
  - Webpack configuration is difficult to reason about
  - Webpack doesn’t build the bundle on Windows
- **Solution:** Rewrite _webpack.config.js_ to use `path` everywhere, and clean it up.
- **Testing:**
  
  From _coinstac/packages/coinstac-ui/_:
  - Run in develop mode:
    
    ``` shell
    export NODE_ENV=development
    node webpack-dev-server.js & node_modules/.bin/electron .
    ```
  - Run in production mode:
    
    ``` shell
    export NODE_ENV=production; npm start
    ```

### CODE REVIEW

File: packages/coinstac-ui/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "react-bootstrap": "^0.30.0",
     "react-dom": "^15.1.0",
     "react-notification-system": "^0.2.7",
+    "react-redux": "^4.4.5",
```
Comment (swashcap) [inline packages/coinstac-ui/package.json:26]: Not sure how we fared without this…

File: packages/coinstac-ui/app/render/components/app.js [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
-      <ul id="loading_list">
-        {Object.keys(props.loading.wip).map((loadingKey, ndx) => {
-          return (<li key={ndx}>{loadingKey}</li>);
-        })}
-      </ul>
-
-      <Notify />
-    </div>
-  );
+class App extends Component { // eslint-disable-line react/prefer-stateless-function
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/app.js:5]: There’s some bug with _react-hot-loader_ not liking functional components. We can revert when that plugin hits v3.

File: packages/coinstac-ui/app/main/utils/boot/configure-core.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
 const app = require('ampersand-app');
 const common = require('coinstac-common');
 const CoinstacClientCore = require('coinstac-client-core');
-const merge = require('lodash/merge');
+const { merge } = require('lodash');
```
Comment (swashcap) [inline packages/coinstac-ui/app/main/utils/boot/configure-core.js:6]: This is mostly to make Webpack’s `externals` configuration easier. It should allow for module tree-shaking in our render process’s bundle once Webpack v2 is released.

File: packages/coinstac-ui/package.json (hunk 1 above)
Comment (rssk) [inline packages/coinstac-ui/package.json:26]: wat

**Pull request merged**